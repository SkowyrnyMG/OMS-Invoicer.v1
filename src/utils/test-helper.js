import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
