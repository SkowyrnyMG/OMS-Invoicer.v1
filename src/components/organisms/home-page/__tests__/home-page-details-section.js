import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import HomePageDetailsSection from '../home-page-details-section/home-page-details-section';

describe('HomePageDetailsSection', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<HomePageDetailsSection />);
  });
});
