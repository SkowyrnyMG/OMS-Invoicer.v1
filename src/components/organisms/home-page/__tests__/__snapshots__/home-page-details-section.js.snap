// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageDetailsSection should display in the document with correct styles 1`] = `
.c0 {
  padding: 5rem 0;
  display: grid;
  grid-template-columns: 5rem [start] repeat(2,1fr) [end] 5rem;
  grid-column: 1 / -1 !important;
  grid-gap: 3%;
  grid-row-gap: 5%;
  min-height: 60rem;
  background-color: #446DF644;
  margin: 0;
  padding-top: 10rem;
}

.c0 > * {
  grid-column: start / end;
}

.c6 {
  padding: 5rem 0;
  display: grid;
  grid-template-columns: 5rem [start] repeat(2,1fr) [end] 5rem;
  grid-column: 1 / -1 !important;
  grid-gap: 3%;
  grid-row-gap: 5%;
  min-height: 60rem;
  background-color: #F8F8FF;
  margin: 0;
  padding-top: 10rem;
}

.c6 > * {
  grid-column: start / end;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  grid-column-gap: 10rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  position: relative;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 34px;
}

.c7 {
  position: relative;
  margin: 10rem 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 34px;
}

.c4 {
  position: absolute;
  line-height: 1;
  top: 0;
  left: 0;
  font-size: 180px;
  color: #446DF644;
  -webkit-transform: translate(-100%,-50%);
  -ms-transform: translate(-100%,-50%);
  transform: translate(-100%,-50%);
  z-index: 0;
}

.c4 svg {
  position: absolute;
  width: 70px;
  fill: #446DF6;
  left: 0;
  top: 50%;
  -webkit-transform: translate(-50%,-30%);
  -ms-transform: translate(-50%,-30%);
  transform: translate(-50%,-30%);
}

.c8 {
  width: 70%;
  margin-bottom: 5rem;
}

.c5 {
  text-align: left;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-left: calc(15rem + 3%);
}

.c9 {
  padding: 1rem;
  margin-left: 5rem;
  margin-bottom: 5rem;
}

@media (max-width:680px) {
  .c0 {
    grid-template-columns: [start] 1fr [end];
    padding: 2rem;
  }
}

@media (max-width:680px) {
  .c6 {
    grid-template-columns: [start] 1fr [end];
    padding: 2rem;
  }
}

@media (max-width:1028px) {
  .c1 {
    grid-template-columns: 1fr;
    grid-column-gap: 0;
  }
}

@media (max-width:680px) {
  .c1 {
    position: relative;
    padding-top: 15rem;
  }
}

@media (max-width:680px) {
  .c3 {
    position: static;
  }
}

@media (max-width:680px) {
  .c7 {
    position: static;
  }
}

@media (max-width:680px) {
  .c4 {
    font-size: 180px;
    left: 50%;
    -webkit-transform: translate(-50%,0);
    -ms-transform: translate(-50%,0);
    transform: translate(-50%,0);
  }
}

@media (max-width:680px) {
  .c2 {
    padding-left: 0;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<section>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          <div
            class="c4"
          >
            <svg>
              clients-icon.svg
            </svg>
            1
          </div>
          Customers details in one place
        </h3>
        
      </div>
      <div
        class="c5"
      >
        <p>
          Keep your customers details in one place. Address, VAT number, e-mail address and many more can be stored directly in customer cards!
          <br />
          <br />
          No matter how many customers your database will contain.
          <br />
          <b>
             Quick search 
          </b>
          in customer module gives you possibility to find any of them within a few seconds!
        </p>
      </div>
    </div>
  </div>
  <div
    class="c6"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          <div
            class="c4"
          >
            <svg>
              orders-icon.svg
            </svg>
            2
          </div>
          Take control over your orders
        </h3>
        
      </div>
      <div
        class="c5"
      >
        <p>
          Keep all of your
          <b>
             orders under control! 
          </b>
          <br />
          <br />
          You can now create orders for customers present in your database. Statuses will help with tracking progress on each individual order.
        </p>
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          <div
            class="c4"
          >
            <svg>
              invoice-icon.svg
            </svg>
            3
          </div>
          Easy invoicing
        </h3>
        
      </div>
      <div
        class="c5"
      >
        <p>
          Invoicing has
          <b>
             never been that easy!
          </b>
          <br />
          <br />
          Issue new invoice with just few mouse clicks. OMS invoicer will take all of neccesary data from orders and provide them on your bill! One thing that user has to do by himself is just to accept draft and print the invoice!
        </p>
      </div>
    </div>
  </div>
  <div
    class="c6"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c7"
        >
          <div
            class="c4"
          >
            <svg>
              ok-icon.svg
            </svg>
            4
          </div>
          Instant VIES verification
        </h3>
        <img
          alt="position image"
          class="c8"
          src="homepage-body-1.jpg"
        />
      </div>
      <div
        class="c5"
      >
        <p>
          Adding new customer by hand is just a relic of the past. With OMS you can fetch most important details about your customers directly from VIES database. To do that you just have to provide country prefix and new customer VAT EU number into invoices modal.
          <br />
          <b>
            What benefints do you have using VIES autofil?
          </b>
        </p>
        <ol
          class="c9"
        >
          <li>
            Save your time!
          </li>
          <li>
            Avoid mistakes!
          </li>
          <li>
            You will be always sure that your new customer have active VAT number
          </li>
          <li>
            It is just easier to do!
          </li>
        </ol>
        <span>
          Do not wait! Try it on your own!
        </span>
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          <div
            class="c4"
          >
            <svg>
              cash-icon.svg
            </svg>
            5
          </div>
          Keep your payments under control
        </h3>
        
      </div>
      <div
        class="c5"
      >
        <p>
          Who did not have a problem with
          <b>
             untracked delayed payments? 
          </b>
          <br />
          <br />
          Thanks to invoice module you can mark which invoices are paid! By using search field you can also quickly filter all unpaid invoices to start collect your liabilities!
        </p>
      </div>
    </div>
  </div>
  <div
    class="c6"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c7"
        >
          <div
            class="c4"
          >
            <svg>
              dashboard-icon.svg
            </svg>
            6
          </div>
          Useful dashboard notifications
        </h3>
        <img
          alt="position image"
          class="c8"
          src="homepage-body-2.jpg"
        />
      </div>
      <div
        class="c5"
      >
        <p>
          To wrap everyting in one place our dashbord will display most important informations. For example
        </p>
        <ol
          class="c9"
        >
          <li>
            Unpaid invoices
          </li>
          <li>
            Finished orders without invoice
          </li>
          <li>
            Pending orders
          </li>
          <li>
            Quick app nav
          </li>
        </ol>
        <p>
          .. and much more!
        </p>
      </div>
    </div>
  </div>
</section>
`;
